### PR TITLE
Create enablement package for evaluating enable when expressions

### PR DESCRIPTION
**Pull request by jingtang10**

### CODE REVIEW

File: datacapture/src/test/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluatorTest.kt [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+                    "q1" -> QuestionnaireResponse.Item.getDefaultInstance()
+                    "q2" -> QuestionnaireResponse.Item.getDefaultInstance()
+                    else -> null
+                }
+            }
+        ).isFalse()
+    }
+
+    @Test
+    fun evaluate_allEnableWhens_allSatisfied_shouldReturnTrue() {
```
Comment (deepankarb) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluatorTest.kt:304]: Not sure if we should add more tests to cover scenarios where answer exists for "any" and "all".

**Review by deepankarb — APPROVED** [review]
Approved, with a comment about some test scenarios.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                .addEnableWhen(
+                    Questionnaire.Item.EnableWhen.newBuilder()
+                        .setQuestion(String.newBuilder().setValue("q1"))
+                        .setOperator(
+                            Questionnaire.Item.EnableWhen.OperatorCode.newBuilder()
+                                .setValue(QuestionnaireItemOperatorCode.Value.EXISTS)
+                        )
+                        .setAnswer(
+                            Questionnaire.Item.EnableWhen.AnswerX.newBuilder().apply {
+                                boolean = Boolean.newBuilder().setValue(true).build()
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluatorTest.kt:67]: 😂

File: datacapture/src/main/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluator.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+     * [QuestionnaireResponse.Item] with the `linkId`, or null if there isn't one.
+     *
+     * For example, the questionnaireItem might be
+     */
+    fun evaluate(
+        questionnaireItem: Questionnaire.Item,
+        questionnaireResponseItemRetriever: (linkId: String) -> QuestionnaireResponse.Item?
+    ): Boolean {
+        val enableWhenList = questionnaireItem.enableWhenList
+        if (enableWhenList.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluator.kt]: done. thanks

File: datacapture/src/test/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluatorTest.kt (hunk 1 above)
Comment (jingtang10) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluatorTest.kt:304]: i think this is logically complete -- do you have any specific suggestions?

**Pull request merged**